libbroker/broker/endpoint.cc

```diff
@@ -560,31 +440,23 @@ endpoint::endpoint(configuration config, endpoint_id id) : id_(id) {
   domain_options adaptation{opts.disable_forwarding};
   if (auto sp = caf::get_as<std::string>(cfg, "caf.scheduler.policy");
       sp && *sp == "testing") {
-    core = sys.spawn<core_t>(id_, filter_type{}, clock_.get(), &adaptation,
-                             std::move(conn_ptr));
+    core = sys.spawn<core_t>(registry_, id_, filter_type{}, clock_.get(),
+                             &adaptation, std::move(conn_ptr));
   } else {
-    core = sys.spawn<core_t, caf::detached>(id_, filter_type{}, clock_.get(),
-                                            &adaptation, std::move(conn_ptr));
+    core = sys.spawn<core_t, caf::detached>(registry_, id_, filter_type{},
+                                            clock_.get(), &adaptation,
+                                            std::move(conn_ptr));
   }
   core_ = facade(core);
-  // Spin up a Prometheus actor if configured or an exporter.
+  // Spin up a Prometheus exposer if configured.
   if (auto port = caf::get_as<broker::port>(cfg, "broker.metrics.port")) {
-    auto ptask = std::make_unique<prometheus_http_task>(sys);
-    auto addr = caf::get_or(cfg, "broker.metrics.address", std::string{});
-    if (auto actual_port =
-          ptask->start(port->number(), native(core_),
-                       addr.empty() ? nullptr : addr.c_str())) {
-      BROKER_INFO("expose metrics on port" << *actual_port);
-      telemetry_exporter_ = facade(ptask->telemetry_exporter());
-      background_tasks_.emplace_back(std::move(ptask));
-    } else {
-      BROKER_ERROR("failed to expose metrics:" << actual_port.error());
-    }
-  } else {
-    using exporter_t = internal::metric_exporter_actor;
-    auto params = internal::metric_exporter_params::from(cfg);
-    auto hdl = sys.spawn<exporter_t>(native(core_), std::move(params));
-    telemetry_exporter_ = facade(hdl);
+    auto str = caf::get_or(cfg, "broker.metrics.address", ""s);
+    if (!str.empty())
+      str += ':';
+    str += std::to_string(port->number());
+    BROKER_INFO("expose metrics on" << str);
+    exposer_ = std::make_unique<prometheus::Exposer>(str);
```

Comment:
Is it actually correct to pass the port as `:PORT` here? While they call the [parameter "`bind_address`"](https://github.com/jupp0r/prometheus-cpp/blob/master/pull/src/exposer.cc#L17) the actual consumer of this value [seems to expect just a plain port (or a list of ports)](https://github.com/civetweb/civetweb/blob/d7ba35bbb649209c66e582d5a0244ba988a15159/include/civetweb.h#L433-L438).

If I make the following edit tests seem to behave identically, so we should probably add something exercising this, especially since most of our work is just glueing things together (we'd want to make sure that actually works):

```cpp
  // DEITY, PLEASE GIVE ME THE STRENGTH TO ADD NEWLINES WHERE THEY MAKE SENSE.

  if (auto port = caf::get_as<broker::port>(cfg, "broker.metrics.port")) {
    BROKER_INFO("expose metrics on" << port->number());
    exposer_ = std::make_unique<prometheus::Exposer>(std::to_string(port->number()));
  }

```

In any case, it seems interesting that at least in the way we use their API there is no way to specify and actual endpoint to listen on (e.g., if we have multiple addresses).